mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -331,8 +347,9 @@ def create(self, *args, **kwargs):
     @action(detail=True, methods=["post", "options"])
     def event(self, request, id):
         """Endpoint to receive an event from a provider."""
-        provider_id = request.user.provider_id
-        device = models.Device.objects.filter(provider_id=provider_id, id=id).last()
+        data_broker_id = request.user.provider_id
+        # We do not filter by provider_id here
+        device = models.Device.objects.filter(id=id).last()
```

Comment:
I don't like this `last()`, we should not create an event on a random device

Reply:
I changed it to `device = models.Device.objects.get(pk=id)`